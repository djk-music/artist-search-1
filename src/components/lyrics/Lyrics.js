import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import { getSongLyrics } from '../../services/artistsAndWorks';

export default class Lyrics extends PureComponent {
  state = {
    lyrics: []
  }
  static propTypes = {
    match: PropTypes.object
  }
  componentDidMount() {
    getSongLyrics(this.props.match.params.artist, this.props.match.params.song)
      .then(response => {
        this.setState({ lyrics: response.results });
      });
  }
  render() {
    const { lyrics } = this.state;
    return (
      <Fragment>
        <h2>Song Title</h2>
        <p>{lyrics}</p>
      </Fragment>
    );
  }
}

