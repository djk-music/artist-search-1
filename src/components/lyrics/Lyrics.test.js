import React from 'react';
import renderer from 'react-test-renderer';
import Lyrics  from './Lyrics';
import { MemoryRouter } from 'react-router-dom';

jest.mock('../../services/artistsAndWorks.js');

describe('Lyrics', () => {
  it('matches a snapshot', () => {
    const match = {
      params: {
        artist: 'artist',
        song: '2344'   
      }
    };
    const tree = renderer.create(
      <MemoryRouter><Lyrics match={match}/></MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
