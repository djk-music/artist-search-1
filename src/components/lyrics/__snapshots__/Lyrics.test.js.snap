// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics matches a snapshot 1`] = `
Array [
  <h2>
    Song Title
  </h2>,
  <p />,
]
`;
